src/dataset/domain/useCases/getDatasetsWithCount.ts

```diff
@@ -0,0 +1,13 @@
+import { DatasetRepository } from '../repositories/DatasetRepository'
+import { DatasetPaginationInfo } from '../models/DatasetPaginationInfo'
+import { DatasetsWithCount } from '../models/DatasetsWithCount'
+
+export async function getAllWithCount(
```

Comment:
Sorry for the confusion! I meant to make the change in the repository, not in the use case. That's because the repository method is always used as `datasetRepository.getAllWithCount`, while the use case is directly referred to as `getAllDatasetsWithCount`. So for the use case, we need the 'datasets' part

```suggestion
export async function getAllDatasetsWithCount(
```